### PR TITLE
Release 1.0.0

### PR DESCRIPTION
**Pull request by cdrage**
Release 1.0.0 of Streamlink!

Check out the CHANGELOG.md changes. This is a big one for us!

### CODE REVIEW

**Comment by cdrage** [comment]
:tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: 
:balloon: :balloon: :balloon: :balloon: :balloon: :balloon: :balloon: :balloon: :balloon:
:tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: 

Ping @beardypig @bastimeyer @gravyboat

**Comment by bastimeyer** [comment]
We should make sure to update these dependencies before publishing a new major version:
https://github.com/streamlink/streamlink/network/alerts

**Comment by amurzeau** [comment]
I'm interested about these dependencies requiring upgrade as they might impact dirstributions.
Are they related to python2 only and not python3 ?

Can they be disclosed or you prefer not to ?

**Comment by bastimeyer** [comment]
@gravyboat 
The version ranges were changed in these commits:
e386cfbc5e0ef9e5998c888aa49db21c70ed1a37 (#965)
2b052ad1e32ca31384d8ec3fd6ebde44ba24b112 (#239)

@amurzeau 
I didn't know you don't have access to the alerts list...

1. urllib3
   > Vulnerable versions: < 1.23
   > Patched version: 1.23
   > urllib3 before version 1.23 does not remove the Authorization HTTP header when following a cross-origin redirect (i.e., a redirect that differs in host, port, or scheme). This can allow for credentials in the Authorization header to be exposed to unintended hosts or transmitted in cleartext.
   https://nvd.nist.gov/vuln/detail/CVE-2018-20060
2. requests
   > Vulnerable versions: <= 2.19.1
   > Patched version: 2.20.0
   > The Requests package through 2.19.1 before 2018-09-14 for Python sends an HTTP Authorization header to an http URI upon receiving a same-hostname https-to-http redirect, which makes it easier for remote attackers to discover credentials by sniffing the network.
   https://nvd.nist.gov/vuln/detail/CVE-2018-18074

**Comment by beardypig** [comment]
Generally it's a good idea to pin versions, otherwise you get inconsistent builds (however, we are no consistent in doing this). I believe the versions that were pinned were the latest at the time :) I would just pin them to the latest current version - assuming it works ;)

**Comment by gravyboat** [comment]
@cdrage okay vulnerabilities are patched and the logo issues are taken care of. Can you rebuild the changelog and do the 1.0.0 release? We're all set!

**Comment by cdrage** [comment]
@gravyboat I'm traveling right now with limited internet availability so I'll be unable to do a release until late next week! However, i did a very extensive changelog for 1.0 that you can look at / edit and then deploy :+1:

**Comment by bastimeyer** [comment]
What's the status of this?
We need to publish a new release... There are many plugins which are broken on 0.14.2, even some popular ones, like youtube. People are forced to use the daily builds on Windows since almost half a year (2018-06-29) and on Linux, all you can do is use pip if your distro only packages stable releases. Same on macOS with the homebrew package.

Please don't forget to update the changelog, as there have been a couple of more merges and breaking changes since then, eg. the record parameter (#2152). Thanks!

**Comment by gravyboat** [comment]
@bastimeyer After @cdrage commented I was planning to do it but was lazy in the evening. If he wants to update the changelog and roll a new release that would be great, otherwise I'll try to get to it as soon as I can.

**Comment by cdrage** [comment]
Hey @gravyboat I can do a release tomorrow and update the changelog, are you good with that? :+1:

**Comment by gravyboat** [comment]
@cdrage That sounds good to me!

**Comment by cdrage** [comment]
Ping @beardypig @bastimeyer @gravyboat

If you guys could quickly check-over this PR then we can release 1.0.0!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,215 @@
...
+
+The celebratory release of Streamlink 1.0.0!
+
+*A lot* of hard work has gone into getting Streamlink to where it is. Not only is Streamlink used across multiple applications and platforms, but companies as well. 
+
+Streamlink started from the inaugural [fork of Livestreamer](https://github.com/chrippa/livestreamer/issues/1427) on September 17th, 2016. 
+
+Since then, We've hit multiple milestones:
+
+ - Over 886 PR requests
```
Comment (bastimeyer) [inline CHANGELOG.md]: "PR requests" is redundant :smiley:

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,215 @@
...
+
+Thanks to everyone who've contributed to Streamlink (and our backers)! Without you, we wouldn't be where we are today.
+
+**Without further ado, here are the changes in release 1.0.0:**
+
+  - We have a new icon / logo for Streamlink! (https://github.com/streamlink/streamlink/pull/2165)
+  - Updated dependencies (https://github.com/streamlink/streamlink/pull/2230)
+  - A *ton* of plugin updates. Have a look at [this search query](https://github.com/streamlink/streamlink/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed+plugins.+) for all the recent updates.
+  - You can now provide a custom key URI to override HLS streams (https://github.com/streamlink/streamlink/pull/2139). For example: `--hls-segment-key-uri <URI>`
+  - User agents for API communication has been updated (https://github.com/streamlink/streamlink/pull/2194)
```
Comment (bastimeyer) [inline CHANGELOG.md]: *have been

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,215 @@
...
+  - Process output will no longer show if tty is unavailable (https://github.com/streamlink/streamlink/pull/2090)
+  - We've removed BountySource in favour of our OpenCollective page. If you have any features you'd like to request, please open up an issue with the request and possibly consider backing us!
+  - Improved terminal progress display for wide characters (https://github.com/streamlink/streamlink/pull/2032)
+  - Fixed a bug with dynamic playlists on playback (https://github.com/streamlink/streamlink/pull/2096)
+  - Fixed makeinstaller.sh (https://github.com/streamlink/streamlink/pull/2098)
+  - Old Livestreamer deprecations and API references were removed (https://github.com/streamlink/streamlink/pull/1987)
+  - Dependencies have been updated for Python (https://github.com/streamlink/streamlink/pull/1975)
+  - Newer and more common User-Agents are now used (https://github.com/streamlink/streamlink/pull/1974)
+  - DASH stream bitrate's now round-up to the nearest 10, 100, 1000, etc. (https://github.com/streamlink/streamlink/pull/1995)
```
Comment (bastimeyer) [inline CHANGELOG.md]: no apostrophe here

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,215 @@
...
+  - Updated documentation on issue templates (https://github.com/streamlink/streamlink/pull/1996)
+  -	URL have been added for better processing of HTML tags (https://github.com/streamlink/streamlink/pull/1675)
```
Comment (bastimeyer) [inline CHANGELOG.md]: tabstop instead of space

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,215 @@
...
+  - Fixed sort and prog issue (https://github.com/streamlink/streamlink/pull/1964)
+  - Reformatted issue templates (https://github.com/streamlink/streamlink/pull/1966)
+  - Fixed crashing bug with player-continuous-http option (https://github.com/streamlink/streamlink/pull/2234)
+  - Make sure all dev dependencies (https://github.com/streamlink/streamlink/pull/2235<Paste>)
+	- Recording (-r) has been fixed (https://github.com/streamlink/streamlink/pull/2152)
```
Comment (bastimeyer) [inline CHANGELOG.md]: tabstop instead of space

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,215 @@
...
+  - Old Livestreamer deprecations and API references were removed (https://github.com/streamlink/streamlink/pull/1987)
+  - Dependencies have been updated for Python (https://github.com/streamlink/streamlink/pull/1975)
+  - Newer and more common User-Agents are now used (https://github.com/streamlink/streamlink/pull/1974)
+  - DASH stream bitrate's now round-up to the nearest 10, 100, 1000, etc. (https://github.com/streamlink/streamlink/pull/1995)
+  - Updated documentation on issue templates (https://github.com/streamlink/streamlink/pull/1996)
+  -	URL have been added for better processing of HTML tags (https://github.com/streamlink/streamlink/pull/1675)
+  - Fixed sort and prog issue (https://github.com/streamlink/streamlink/pull/1964)
+  - Reformatted issue templates (https://github.com/streamlink/streamlink/pull/1966)
+  - Fixed crashing bug with player-continuous-http option (https://github.com/streamlink/streamlink/pull/2234)
+  - Make sure all dev dependencies (https://github.com/streamlink/streamlink/pull/2235<Paste>)
```
Comment (bastimeyer) [inline CHANGELOG.md]: Something seems to have gone wrong on this line here

File: CHANGELOG.md (hunk 5 above)
Comment (bastimeyer) [reply CHANGELOG.md]: Oh, and the `-r` parameter wasn't fixed, it has been replaced with the new recording logic (in addition to `-R`). `-r` was previously a shorthand parameter for `--rtmp-rtmpdump`.
https://github.com/streamlink/streamlink/pull/2152#issue-228209657

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,215 @@
...
+Streamlink started from the inaugural [fork of Livestreamer](https://github.com/chrippa/livestreamer/issues/1427) on September 17th, 2016. 
+
+Since then, We've hit multiple milestones:
+
+ - Over 886 PR requests
+ - Hit 3,000 commits in Streamlink
+ - Obtaining our first sponsors as well as backers of the project
+ - The creation of our own logo (https://github.com/streamlink/streamlink/issues/1123)
+
+Thanks to everyone who've contributed to Streamlink (and our backers)! Without you, we wouldn't be where we are today.
```
Comment (gravyboat) [inline CHANGELOG.md]: I think `who has` would work better than `who've` in this context.

**Comment by cdrage** [comment]
@bastimeyer @gravyboat I fixed the grammatical errors! :+1:

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,215 @@
...
+
+The celebratory release of Streamlink 1.0.0!
+
+*A lot* of hard work has gone into getting Streamlink to where it is. Not only is Streamlink used across multiple applications and platforms, but companies as well. 
+
+Streamlink started from the inaugural [fork of Livestreamer](https://github.com/chrippa/livestreamer/issues/1427) on September 17th, 2016. 
+
+Since then, We've hit multiple milestones:
+
+ - Over 886 PR's
```
Comment (beardypig) [inline CHANGELOG.md]: Shouldn't be it PRs (Pull Request**s**)?

```suggestion
 - Over 886 PRs
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,215 @@
...
+  - Dependencies have been updated for Python (https://github.com/streamlink/streamlink/pull/1975)
+  - Newer and more common User-Agents are now used (https://github.com/streamlink/streamlink/pull/1974)
+  - DASH stream bitrates now round-up to the nearest 10, 100, 1000, etc. (https://github.com/streamlink/streamlink/pull/1995)
+  - Updated documentation on issue templates (https://github.com/streamlink/streamlink/pull/1996)
+  - URL have been added for better processing of HTML tags (https://github.com/streamlink/streamlink/pull/1675)
+  - Fixed sort and prog issue (https://github.com/streamlink/streamlink/pull/1964)
+  - Reformatted issue templates (https://github.com/streamlink/streamlink/pull/1966)
+  - Fixed crashing bug with player-continuous-http option (https://github.com/streamlink/streamlink/pull/2234)
+  - Make sure all dev dependencies (https://github.com/streamlink/streamlink/pull/2235)
+	- Recording (-r) has been replaced with new recording logic(https://github.com/streamlink/streamlink/pull/2152)
```
Comment (beardypig) [inline CHANGELOG.md]:
```suggestion
	- Recording (-r) has been replaced with new recording logic (https://github.com/streamlink/streamlink/pull/2152)
```

Nitpicking, space added ...

Comment (back-to) [reply CHANGELOG.md]: there is a tab at the start

**Comment by cdrage** [comment]
@bastimeyer I blame the canadian winter haha. Done :+1:

**Comment by cdrage** [comment]
@bastimeyer Just rebased it!

**Comment by bastimeyer** [comment]
I meant the git shortlog at the bottom which is out of date...

**Comment by cdrage** [comment]
@bastimeyer Woops. Updated!

**Comment by gravyboat** [comment]
Looks great, thanks @cdrage!

**Comment by amurzeau** [comment]
Isn't it ok to merge now everything is approved ?

**Comment by gravyboat** [comment]
@cdrage Everyone has approved and I've merged in #2276, if you want to rebase and release it should be good!

**Comment by cdrage** [comment]
Rebased! Let's release :+1:

**Comment by gravyboat** [comment]
@cdrage Do you want me to merge or are you going to self merge and do the release on the github page at the same time?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,246 @@
 # Changelog
 
+## streamlink 1.0.0 (2019-01-22)
```
Comment (back-to) [inline CHANGELOG.md]: update the **date** before you actually merge it
```suggestion
## streamlink 1.0.0 (2019-01-30)
```

**Comment by cdrage** [comment]
@gravyboat If you don't mind doing the release today that'd be great. Unfortunately real-life keeps getting in the way this week (the heavy snow in Canada is really affecting my plane flights lol)

**Comment by gravyboat** [comment]
@cdrage Do you have the instructions around for the required steps for a full release? I can't remember what they are.

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,246 @@
...
+  - Dependencies have been updated for Python (https://github.com/streamlink/streamlink/pull/1975)
+  - Newer and more common User-Agents are now used (https://github.com/streamlink/streamlink/pull/1974)
+  - DASH stream bitrates now round-up to the nearest 10, 100, 1000, etc. (https://github.com/streamlink/streamlink/pull/1995)
+  - Updated documentation on issue templates (https://github.com/streamlink/streamlink/pull/1996)
+  - URL have been added for better processing of HTML tags (https://github.com/streamlink/streamlink/pull/1675)
+  - Fixed sort and prog issue (https://github.com/streamlink/streamlink/pull/1964)
+  - Reformatted issue templates (https://github.com/streamlink/streamlink/pull/1966)
+  - Fixed crashing bug with player-continuous-http option (https://github.com/streamlink/streamlink/pull/2234)
+  - Make sure all dev dependencies (https://github.com/streamlink/streamlink/pull/2235)
+  - Recording (-r) has been replaced with new recording logic(https://github.com/streamlink/streamlink/pull/2152)
```
Comment (qkolj) [inline CHANGELOG.md]: This line is not accurate. #2152 removed `-r` as a short option for `--rtmp-rtmpdump` argument, and repurposed it for new recording functionality (along with `-R`, `-o` and `-O`).

**Comment by amurzeau** [comment]
If the pip release can be done within next week-end, maybe I can manage to get this super-version into the next Debian stable release before the freeze on 12/02.
I hope this is still doable despite the environmental forces going against ^^ courage !

**Comment by cdrage** [comment]
@gravyboat Pretty sure it's just merge it in and it'll auto-create.

**Comment by cdrage** [comment]
I'm going to just go ahead and merge. Too much discussion! If something goes wrong let's update it with a PR. Mergin'

**Comment by cdrage** [comment]
@beardypig I created the tag, just waiting on Travis :) https://github.com/streamlink/streamlink/releases

**Comment by gravyboat** [comment]
Great stuff @cdrage, thanks for geting the release pushed out, I forgot that it was that easy now :)

**Pull request merged**